osmoutils/store_helper_test.go

```diff
@@ -186,3 +189,220 @@ func (s *TestSuite) TestNoStopFn_AlwaysFalse() {
+
+			expectPanic: true,
+		},
+	}
+
+	for name, tc := range tests {
+		s.Run(name, func() {
+			s.SetupStoreWithBasePrefix()
+
+			// Setup
+			for key, value := range tc.preSetKeyValues {
+				osmoutils.MustSet(s.store, []byte(key), value)
+			}
+
+			osmoassert.ConditionalPanic(s.T(), tc.expectPanic, func() {
+				for key, expectedValue := range tc.getKeyValues {
+					// System under test.
+					osmoutils.MustGet(s.store, []byte(key), tc.actualResultProto)
+					// Assertions.
+					s.Require().Equal(expectedValue.String(), tc.actualResultProto.String())
+				}
+			})
+		})
+	}
+}
+
+// TestMustSet tests that MustSet updates the store correctly
+// and panics if an error is encountered.
+func (s *TestSuite) TestMustSet() {
+	tests := map[string]struct {
+		// keyys and values to preset
+		setKey   string
+		setValue proto.Message
+
+		// keys and values to attempt to get and validate
+		getKeyValues map[string]proto.Message
+
+		actualResultProto proto.Message
+
+		key         []byte
+		result      proto.Message
+		expectPanic bool
+	}{
+		"basic valid Dec test": {
+			setKey: keyA,
+			setValue: &sdk.DecProto{
+				Dec: sdk.OneDec(),
+			},
+
+			actualResultProto: &sdk.DecProto{},
+		},
+		"basic valid TwapRecord test": {
+			setKey: keyA,
+			setValue: &twaptypes.TwapRecord{
+				PoolId: 2,
+			},
+
+			actualResultProto: &twaptypes.TwapRecord{},
+		},
+		"invalid set value": {
+			setKey:   keyA,
+			setValue: (*sdk.DecProto)(nil),
+
+			expectPanic: true,
+		},
+	}
+
+	for name, tc := range tests {
+		s.Run(name, func() {
+			s.SetupStoreWithBasePrefix()
+
+			// Setup
+			osmoassert.ConditionalPanic(s.T(), tc.expectPanic, func() {
+				osmoutils.MustSet(s.store, []byte(tc.setKey), tc.setValue)
+			})
+
+			if tc.expectPanic {
+				return
+			}
+
+			osmoutils.MustGet(s.store, []byte(tc.setKey), tc.actualResultProto)
+			s.Require().Equal(tc.setValue.String(), tc.actualResultProto.String())
+		})
+	}
+}
+
+// TestMustGetDec tests that MustGetDec retrieves the correct
+// decimal values from the store and panics if an error is encountered.
+func (s *TestSuite) TestMustGetDec() {
+
+	tests := map[string]struct {
+		// keyys and values to preset
+		preSetKeyValues map[string]sdk.Dec
+
+		// keys and values to attempt to get and validate
+		getKeyValues map[string]sdk.Dec
+
+		expectPanic bool
+	}{
+		"valid get": {
+			preSetKeyValues: map[string]sdk.Dec{
+				keyA: sdk.OneDec(),
+				keyB: sdk.OneDec().Add(sdk.OneDec()),
+				keyC: sdk.OneDec().Add(sdk.OneDec()).Add(sdk.OneDec()),
+			},
+
+			getKeyValues: map[string]sdk.Dec{
+				keyA: sdk.OneDec(),
+				keyB: sdk.OneDec().Add(sdk.OneDec()),
+				keyC: sdk.OneDec().Add(sdk.OneDec()).Add(sdk.OneDec()),
+			},
+		},
+		"attempt to get non-existent key - panic": {
+			preSetKeyValues: map[string]sdk.Dec{
+				keyA: sdk.OneDec(),
+				keyC: sdk.OneDec().Add(sdk.OneDec()).Add(sdk.OneDec()),
+			},
+
+			getKeyValues: map[string]sdk.Dec{
+				keyB: sdk.OneDec().Add(sdk.OneDec()),
+			},
```

Comment:
Can we modify this so that the getKeyValues contains both an existing and non existing value (i.e. just add keyA to getKeyValues?) That or leave this be and add another test?